design-docs/handling-encodings.md

```diff
@@ -0,0 +1,120 @@
+[Pull Request 499](https://github.com/gradle/gradle/pull/499) which allows specifying a metadata charset, which defaults to
+the platform default encoding. The name of this option is confusing, and the default value is not correct. Moreover, jar, war and ear
+metadata should always be encoded in UTF8 instead of the platform default encoding. The Tar task should use UTF8, but it
+uses the default platform encoding.
+
+## Solution
+
+### Content encoding
+
+This design generally proposes that any task which needs to read and write text should specify a charset used to encode/decode 
+characters into bytes.
+
+Notes:
+
+ - reading and writing characters isn't limited to files. It also applies to archive entries, network communication, etc.
+ - some tasks might need to use a charset to write that is different from the charset used to read. For the most frequent case,
+   i.e. copying and filtering files, it's a reasonable assumption to expect the developer to choose the target encoding for the
+   source files, and thus to use a single filtering charset both for reading and writing. By the way, when not filtering the 
+   files, the Copy task treats them as binary files, and thus uses the same encoding for the source and target files.
+
+### Metadata encoding
+
+The Zip task, which produces zip files, should allow specifying a metadata charset used to encode/decode the metadata
+of the archive. This should default to UTF8. 
+
+Jar, War, Ear and Tar tasks should not allow specifying a metadata charset, or at the very least should reject any metadata 
+charset other than UTF8.
+
+When reading from zip files using `ZipFileTree`, we should support a metadata charset option as well.
+
+When reading from tar files using `TarFileTree`, we should support a metadata charset option so that we can handle reading 
+non-POSIX tar files, like the tar files created with current and older versions of Gradle, or with other tools.
+
+## Implementation Plan
+
+### Content encoding
+
+`CopySpec` should get a new `filteringCharset` property of type `String` which is the name of the `Charset` which will be used for 
+encoding and decoding the contents of a file on either side of passing the contents to the closure(s) specified by the various 
+filter methods of CopySpec. 
+
+The default value of `filteringCharset` should be the name of the platform default charset. That will maintain backward 
+compatibility, since up to Gradle 1.12 at least, the platform default charset is always used.
+
+### Metadata encoding
+
+The Zip task should support a `metadataCharset` option, or type String, defaulting to UTF8. Unfortunately, the Jar task 
+inherits from Zip, and should not use any `metadataCharset` other than UTF8. The setter of this property should thus be 
+overridden to throw an ÌnsupportedOperationException`. War and Ear inherit from Jar, and will thus automatically benefit
```

Comment:
That's a minor detail but the markup is broken here.
It should be `UnsupportedOperationException`.
Instead of **Ì**nsupportedOperationException`.